src/screens/RegistrationScreen/ClientRegistrationScreen.tsx

```diff
@@ -24,256 +25,172 @@ import {
+	const { navigate, goBack } = useNavigation();
 	const [ _state, actions ] = useGlobal() as any;
 	const { register } = actions;
 
-	const transportationMethods = [
-		'Walk',
-		'Bike',
-		'Public',
-		'Car',
-	];
-
-	const ethnicities = [
-		'Prefer not to say',
-		'American Indian or Alaskan Native',
-		'Asian',
-		'Black or African American',
-		'Hispanic or Latino',
-		'Native Hawaiian/Pacific Islander',
-		'White',
-	];
-
-	const genders = [
-		'Prefer not to say',
-		'Female',
-		'Male',
-		'Non-binary',
-		'Other',
-	];
-
-	const itemizeList = list => list.map(item => ({ label: item, value: item }));
-
-	const ChevronDown = () => (
-		<View style={styles.iconContainer}>
-			<Icon
-				name="chevron-down"
-				size={24}
-			/>
-		</View>
-	);
-
-	const selectPlaceholder = {
-		label: 'Select...',
-		color: 'gray',
-		value: null,
-	};
-
-	const [ hidePwd, setHidePwd ] = useState(true);
 	const [ email, setEmail ] = useState('');
 	const [ password, setPassword ] = useState('');
-	const [ street, setStreet ] = useState('');
-	const [ city, setCity ] = useState('');
-	const [ state, setState ] = useState('WA');
-	const [ zip, setZip ] = useState('');
-	const [ transportationMethod, setTransportationMethod ] = useState();
-	const [ gender, setGender ] = useState();
-	const [ ethnicity, setEthnicity ] = useState();
+	const [ retypedPassword, setRetypedPassword ] = useState('');
 	const [ termsOfService, setTermsOfService ] = useState(false);
+	const [ firstName, setFirstName ] = useState('');
+	const [ lastName, setLastName ] = useState('');
 
 	const toggleTermsOfService = () => {
 		setTermsOfService(!termsOfService);
 	};
 
-	const validateAndSubmit = async () => {
-		if (!email.includes('@') || !email.includes('.')) { Alert.alert('Please enter a valid email address.'); return; }
-		if (password.length < 8) { Alert.alert('Please enter a password at least 8 characters long.'); return; }
-		if (!street || street.split(' ').length < 3) { Alert.alert('Please enter your street number and name.'); return; }
-		if (!city) { Alert.alert('Please enter your city.'); return; }
-		if (!(/^\d{5}$/.test(zip))) { Alert.alert('Please enter a valid 5-digit zip code.'); return; }
-		if (!transportationMethod) { Alert.alert('Please select your preferred method of transportation.'); return; }
-		if (!termsOfService) { Alert.alert('Please read and accept the terms of service to complete your registration.'); return; }
+	// object contains the error status and corresponding message
+	// modifications in errorObj trigger change in UI.
+	const errorObj = {
+		email: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		password: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		retypedPassword: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		firstName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+		lastName: {
+			isTrue: false,
+			message: 'Unknown error occurred',
+		},
+	};
```

Comment:
Hey Tao, I'm probably missing some context, but what's your plan for `errorObj`?  The property values are currently the same for every field.  Is your intention to do all the validation on the rails side and have it return this error object to the client or are you planning to do server and client validation?  I'm happy to help on the rails side so that we could possibly have things working end to end when you merge.

Reply:
emmm, neither. My intention to set the `errorObj` there is to create an interface between error information sent back by validation and inputs components on the screen.  For example, if validation find length of password smaller than 8, it could modify `password` attribute under `errorObj` and pass error message to it, and it would trigger corresponding UI change on the screen.